### PR TITLE
Removed VFC

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- removed usage of VFC since it will be removed in next kotlin-wrappers release

It's part of #2221 (extracted to reduce a huge PR)

### CODE REVIEW

**Review by kgevorkyan — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RouterUtils.kt [hunk 1]
```diff
@@ -44,7 +44,7 @@ fun ChildrenBuilder.withNavigate(handler: ChildrenBuilder.(NavigateFunctionConte
  * @return a function component
  */
 @Suppress("TYPE_ALIAS")
-fun <T : Props> withRouter(handler: ChildrenBuilder.(Location, Params) -> Unit) = FC<T> {
+fun withRouter(handler: ChildrenBuilder.(Location, Params) -> Unit) = FC<Props> {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RouterUtils.kt:47]:
```suggestion
fun withRouter(handler: ChildrenBuilder.(Location, Params) -> Unit) = FC {
```

**Review by sanyavertolet — APPROVED** [review]

**Review by Cheshiriks — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/App.kt [hunk 2]
```diff
@@ -33,7 +33,7 @@ import kotlinx.serialization.json.Json
 @JsExport
 @OptIn(ExperimentalJsExport::class)
 @Suppress("VARIABLE_NAME_INCORRECT_FORMAT", "NULLABLE_PROPERTY_TYPE", "EMPTY_BLOCK_STRUCTURE_ERROR")
-val App: VFC = FC {
+val App: FC<Props> = FC {
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/App.kt:36]: sure, my bad, nevermind

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RouterUtils.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RouterUtils.kt:47]: Will also be done in later pr, I guess

Comment (nulls) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RouterUtils.kt:47]: it doesn't work in old version.
I can update it now to 
```
fun withRouter(handler: ChildrenBuilder.(Location, Params) -> Unit): FC<Props> = FC {
```
but it (FC<Props>) will be removed after upgrading

Comment (nulls) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RouterUtils.kt:47]: will do after upgrading together with removing `: FC<Props> = FC {`

Comment (nulls) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RouterUtils.kt]: updated

**Review by DrAlexD — APPROVED** [review]

**Comment by akuleshov7** [comment]
I remember when we moved to VFC, everywhere it was written that VFC is our future and it is only CORRECT way to have it. Now we are removing it, I am pretty sure in 1-2 years we will refactor it once again...

**Review by akuleshov7 — APPROVED** [review]

**Comment by sanyavertolet** [comment]
> I remember when we moved to VFC, everywhere it was written that VFC is our future and it is only CORRECT way to have it. Now we are removing it, I am pretty sure in 1-2 years we will refactor it once again...

I am still waiting for a day, when `ChildrenBuilder` will be removed
screenshot

**Pull request merged**